src/app/courses/courses/courses.component.ts

```diff
@@ -14,14 +16,19 @@ export class CoursesComponent {
 
   deleteDialogRef: MatDialogRef<CoursesDeleteDialogComponent>;
 
-  constructor(private coursesService: CoursesService, private dialog: MatDialog) {
-  }
+  constructor(private authService: AuthService, private router: Router, private coursesService: CoursesService, private dialog: MatDialog) { }
 
   searchCourseByTitle(event: string) {
     this.coursesService.setFilter(event);
   }
   
   deleteCourse(item: CoursesListItem) {
+    if (!this.authService.isAuthenticated()) { 
+      this.authService.setRedirectUrl('/courses');
```

Comment:
I'd personally create an enum with all possible routes and use them in the app, because there are plenty places where you redirect to courses page or login page, if it changes in the future you will have to change in many different places, with enum you could change only one line of code